sdk/core/azure-core/src/main/java/com/azure/core/implementation/jackson/ObjectMapperShim.java

```diff
@@ -275,18 +283,40 @@ private JavaType createJavaType(Type type) {
                 javaTypeArguments[i] = createJavaType(actualTypeArguments[i]);
             }
 
-            return getFromCache(type, t -> mapper.getTypeFactory()
+            return getFromCache(TYPE_TO_JAVA_TYPE_CACHE, type, t -> mapper.getTypeFactory()
                 .constructParametricType((Class<?>) parameterizedType.getRawType(), javaTypeArguments));
         } else {
-            return getFromCache(type, t -> mapper.getTypeFactory().constructType(t));
+            return getFromCache(TYPE_TO_JAVA_TYPE_CACHE, type, t -> mapper.getTypeFactory().constructType(t));
         }
     }
 
+    @SuppressWarnings("unchecked")
     public <T> T deserialize(HttpHeaders headers, Type deserializedHeadersType) throws IOException {
         if (deserializedHeadersType == null) {
             return null;
         }
 
+        try {
+            Class<?> headersClass = TypeUtil.getRawClass(deserializedHeadersType);
+
+            MethodHandle constructor = getFromCache(TYPE_TO_STRONGLY_TYPED_HEADERS_CONSTRUCTOR_CACHE,
+                deserializedHeadersType, type -> {
+                    try {
+                        MethodHandles.Lookup lookup = ReflectionUtilsApi.INSTANCE.getLookupToUse(headersClass);
+                        return lookup.unreflectConstructor(headersClass.getDeclaredConstructor(HttpHeaders.class));
+                    } catch (Throwable throwable) {
+                        return null;
```

Comment:
when would this happen ? is it ok to catch throwable here (could be OOM)?

Reply:
It's that `getLookupToUse` throws `Throwable` (as the APIs used internally throw `Throwable`), we could let this go uncaught depending on the thrown error type